presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -552,7 +553,7 @@ private synchronized void setQueryOutputInfo(QueryExecution.QueryOutputInfo outp
         }
 
         for (URI outputLocation : outputInfo.getBufferLocations()) {
-            exchangeClient.addLocation(outputLocation);
+            exchangeClient.addLocation(outputLocation, TaskId.valueOf("taskid"));
```

Comment:
We can make this magic constant a static final and give a proper name, otherwise it's a bit confusing to see a hardcoded task id here.

Reply:
@nezihyigitbasi : Thanks for the catch!I incorrectly use `TaskIdvalueOf()` here, since I add too much such cases for tests. Here we should pass in the real TaskId. This is done by new added commit "Pass remoteSourceTaskId in ExchangeClient#addLocation"